src/extensions/default/bramble/lib/XHRManager.js

```diff
@@ -0,0 +1,61 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var UrlCache         = brackets.getModule("filesystem/impls/filer/UrlCache");
+    var Path             = brackets.getModule("filesystem/impls/filer/BracketsFiler").Path;
+    var fs               = brackets.getModule("filesystem/impls/filer/BracketsFiler").fs();
+    var Browser          = require("lib/iframe-browser");
+    var transport        = require("lib/PostMessageTransport");
+
+    var XHRManagerRemote = require("text!lib/XHRManagerRemote.js");
+
+    function getRemoteScript() {
+        // Intercept XHR requests, but only if we're rewriting URLs (e.g., Blob URLs)
+        if(UrlCache.getShouldRewriteUrls()) {
+            return "<script>\n" + XHRManagerRemote + "</script>\n";
+        }
+        return "";
+    }
+
+    function getNavigationPath(message) {
```

Comment:
How is this different than the next function `isXHRRequest`? Seems to return the same output, no?